### ISSUE TITLE
"Buy now" button

### ISSUE THREAD

**Issue by rodrigoprimo** [woocommerce/woocommerce#20491]
**Is your feature request related to a problem? Please describe.**
"Buy now" buttons are somewhat popular among e-commerce solutions. WooCommerce.com itself recently added a "Buy now" button to product pages.

In a first iteration, if a user clicked the button, they would skip the cart page and be taken directly to checkout. In the current version, users on WooCommerce.com can complete their purchase straight from the product page. When they click the "Buy now" button, an AJAX call loads the checkout form in the product page.

The problem is that it is not easy to implement this solution using WC core. As far as I can tell, it is necessary to trick WC by faking a user going through the checkout page in a browser. This solution is hacky and not future proof.

**Describe the solution you'd like**
WC core should either provide a built-in "Buy now" button or make it easier for third-party code to create one. I'm not sure if it should support both a button that skips the cart page and a button that loads the checkout form in the product page or just one of those solutions.

**Comment by woocommercebot** [comment]
Thanks for the suggestion @rodrigoprimo.

We'll keep an eye on the [popularity of this request](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md#feature-requests) :)

**Comment by erikdemarco** [comment]
I agree. Look at 10 most popular marketplace/ecommerce. All of them have "buy now" button in product page.

Currently all woocommerce "buy now" plugin is useless or cheaply made without going thru research.

This is how real "Buy now" button should work:
-) when customer click "buy now". It will clear all current cart. And continue add the product. 
-) when customer click "add to cart". Revert the previously deleted cart. And continue add the product.

This is how ALL woocommerce "Buy now" plugin work:
-) when customer click "buy now". It will add product to cart. It don clear the current cart first. 
-) when customer click "add to cart". A usual "add to cart" happen. 

See the difference? All big guy on ecommerce doing the "Buy now" button like I explain in the first section. 
But all woocommerce developer is very lazy and took the easy path.

**Comment by erikdemarco** [comment]
After days searching. I found the 'almost' perfect "buy now" button for woocommerce, although its cant fulfill what i wanted.

For anyone searching this. I found solution from @lomars but still buggy here:
https://stackoverflow.com/a/54867369/15185328

For example if the variation is not selected yet, it still redirecting user. The correct way is. it should alert the user first like the default what woocommerce done. Hopefully @lomars will fix it.
